src/open_inwoner/cms/cases/views/cases.py

```diff
@@ -55,17 +71,33 @@ class InnerCaseListView(
     def page_title(self):
         return _("Mijn aanvragen")
 
-    def get_cases(self):
-        client = build_zaken_client()
-
-        if client is None:
-            return []
-
-        raw_cases = client.fetch_cases(**get_user_fetch_parameters(self.request))
-
-        preprocessed_cases = preprocess_data(raw_cases)
+    def get_cases_for_api_group(self, group: ZGWApiGroupConfig):
+        zaken_client = group.zaken_client
+        raw_cases = zaken_client.fetch_cases(**get_user_fetch_parameters(self.request))
+        preprocessed_cases = preprocess_data(raw_cases, group)
         return preprocessed_cases
 
+    def get_cases(self) -> list[ZaakWithApiGroup]:
+        all_api_groups = list(ZGWApiGroupConfig.objects.all())
+        with parallel() as executor:
+            futures = [
+                executor.submit(self.get_cases_for_api_group, group)
+                for group in all_api_groups
+            ]
+
+            cases = []
+            for task in concurrent.futures.as_completed(futures):
+                try:
+                    task_group = all_api_groups[futures.index(task)]
```

Comment:
Maybe it's because I'm coming from my celery PR, but this variable name sounds strange (I think of `task_group` as a group of (celery) tasks). Perhaps `task_api_group`?